### PR TITLE
handles IB specific flow (Account activation in progress) 

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

File: TradeItIosEmsApi/TradeItOAuthAccessTokenResult.h [hunk 1]
```diff
@@ -3,5 +3,6 @@
 @interface TradeItOAuthAccessTokenResult: TradeItAuthLinkResult
 
 @property (nullable) NSString *broker;
+@property (nullable) NSString *activationTime; // IMMEDIATE if there is no delay using the linked account, otherwise: ONE_OR_TWO_BUSINESS_DAY
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItOAuthAccessTokenResult.h]: What happens if another broker comes along that is 1 week? Is that likely to happen? @TheAntonioReyes

File: TradeItIosTicketSDK2/TradeItAccountSelectionViewController.swift [hunk 2]
```diff
@@ -23,24 +24,24 @@ class TradeItAccountSelectionViewController: CloseableViewController, TradeItAcc
         super.viewWillAppear(animated)
         
         self.promptLabel.text = promptText ?? "SELECT AN ACCOUNT FOR TRADING"
-        let enabledBrokers = TradeItSDK.linkedBrokerManager.getAllEnabledLinkedBrokers()
-        self.accountSelectionTableManager.updateLinkedBrokers(withLinkedBrokers: enabledBrokers, withSelectedLinkedBrokerAccount: selectedLinkedBrokerAccount)
-        if enabledBrokers.isEmpty {
+        let linkedBrokers = TradeItSDK.linkedBrokerManager.getAllEnabledAndActivationInProgressLinkedBrokers()
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItAccountSelectionViewController.swift]: This isn't a criticism of this code, but I'm starting to worry we have too many "states" to keep in mind. I will have no idea which one to use in the future. Can we think of an aggregate term for like `getAllDisplayableLinkedBrokers`? Thoughts @mitochondrion?

File: TradeItIosTicketSDK2/TradeItLinkedBrokerCache.swift [hunk 3]
```diff
@@ -51,6 +52,10 @@ class TradeItLinkedBrokerCache {
         }
 
         linkedBroker.accountsLastUpdated = serializedLinkedBroker[ACCOUNTS_LAST_UPDATED_KEY] as? Date
+        linkedBroker.isAccountLinkDelayedError = serializedLinkedBroker[ACCOUNTS_LINK_DELAY_ERROR_KEY] as? Bool ?? false
+        if linkedBroker.isAccountLinkDelayedError {
+            linkedBroker.error = TradeItErrorResult(title: "Activation In Progress", message: "Your \(linkedBroker.brokerName) is being activated. Check back soon (ip to two business days)", code: TradeItErrorCode.accountNotAvailable)
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerCache.swift]: Typo

File: TradeItIosTicketSDK2/TradeItPortfolioAccountsTableViewManager.swift [hunk 4]
```diff
@@ -203,10 +203,13 @@ fileprivate class LinkedBrokerSectionPresenter {
     private func errorCell(forTableView tableView: UITableView) -> UITableViewCell {
         guard let error = error else { return UITableViewCell() }
         let cell = tableView.dequeueReusableCell(withIdentifier: "TRADE_IT_PORTFOLIO_LINKED_BROKER_ERROR") ?? UITableViewCell(style: .subtitle, reuseIdentifier: nil)
-        if error.requiresRelink() == true {
+        if error.isAccountLinkDelayedError() {
+            cell.textLabel?.text = "Activation in Progress"
+            cell.detailTextLabel?.text = "Account is being activated. Tap to retry."
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItPortfolioAccountsTableViewManager.swift]: Let's review the wording here. If it is going to take 1-2 days then the user might get confused by telling them to `Tap to retry`.

Comment (TheAntonioReyes) [inline TradeItIosTicketSDK2/TradeItPortfolioAccountsTableViewManager.swift]: Agreed, can we change it to "Check back soon."

File: TradeItIosEmsApi/TradeItOAuthAccessTokenResult.h (hunk 1 above)
Comment (TheAntonioReyes) [inline TradeItIosEmsApi/TradeItOAuthAccessTokenResult.h]: then we'll add to the enum, seems very unlikely though

**Pull request merged**